multisig/simulations/ethereum/vip-042/index.ts

```diff
@@ -0,0 +1,160 @@
+  },
+  {
+    market: LST_vwstETH,
+    supply: parseUnits("120", 18),
+    borrow: parseUnits("0", 18),
+    distributor: LST_XVS_DISTRIBUTOR,
+  },
+];
+
+forking(20228802, async () => {
+  describe("Post-Execution state", () => {
+    let xvs: Contract;
+    let prevCoreDistributorBalance: BigNumber;
+    let prevCurveDistributorBalance: BigNumber;
+    let prevLstDistributorBalance: BigNumber;
+
+    before(async () => {
+      xvs = await ethers.getContractAt(XVS_ABI, XVS);
+
+      prevCoreDistributorBalance = await xvs.balanceOf(CORE_XVS_DISTRIBUTOR);
+      prevCurveDistributorBalance = await xvs.balanceOf(CURVE_XVS_DISTRIBUTOR);
+      prevLstDistributorBalance = await xvs.balanceOf(LST_XVS_DISTRIBUTOR);
+
+      await pretendExecutingVip(await vip042());
+    });
+
+    it("registers core distributor in the core comptroller", async () => {
+      const coreComptroller = await ethers.getContractAt(COMPTROLLER_ABI, CORE_COMPTROLLER);
+      const coreDistributors = await coreComptroller.getRewardDistributors();
+      expect(coreDistributors).to.have.lengthOf(3);
+      expect(coreDistributors[2]).to.equal(CORE_XVS_DISTRIBUTOR);
+    });
+
+    it("registers Curve distributor in the Curve comptroller", async () => {
+      const curveComptroller = await ethers.getContractAt(COMPTROLLER_ABI, CURVE_COMPTROLLER);
+      const curveDistributors = await curveComptroller.getRewardDistributors();
+      expect(curveDistributors).to.have.lengthOf(3);
+      expect(curveDistributors[2]).to.equal(CURVE_XVS_DISTRIBUTOR);
+    });
+
+    it("registers LST distributor in the LST comptroller", async () => {
+      const lstComptroller = await ethers.getContractAt(COMPTROLLER_ABI, LST_COMPTROLLER);
+      const lstDistributors = await lstComptroller.getRewardDistributors();
+      expect(lstDistributors).to.have.lengthOf(4);
+      expect(lstDistributors[3]).to.equal(LST_XVS_DISTRIBUTOR);
+    });
+
+    for (const rewardsDistributorAddress of [CORE_XVS_DISTRIBUTOR, CURVE_XVS_DISTRIBUTOR, LST_XVS_DISTRIBUTOR]) {
+      it(`accepts ownership over ${rewardsDistributorAddress}`, async () => {
+        const rewardsDistributor = await ethers.getContractAt(REWARDS_DISTRIBUTOR_ABI, rewardsDistributorAddress);
+        expect(await rewardsDistributor.owner()).to.equal(NETWORK_ADDRESSES.ethereum.NORMAL_TIMELOCK);
+      });
+    }
+
+    it("has correct daily speeds", async () => {
+      for (const data of DAILY_REWARDS) {
+        const { supply, borrow, distributor, market } = data;
+        const rewardsDistributor = await ethers.getContractAt(REWARDS_DISTRIBUTOR_ABI, distributor);
+
+        const supplySpeed = await rewardsDistributor.rewardTokenSupplySpeeds(market);
+        const borrowSpeed = await rewardsDistributor.rewardTokenBorrowSpeeds(market);
+
+        expect(supplySpeed).to.be.closeTo(supply.div(BLOCKS_IN_ONE_DAY), parseUnits("0.01", 18));
+        expect(borrowSpeed).to.be.closeTo(borrow.div(BLOCKS_IN_ONE_DAY), parseUnits("0.01", 18));
```

Comment:
Nit: Instead of these tests, we can use [this](https://github.com/VenusProtocol/vips/blob/main/src/vip-framework/checks/rewardsDistributor.ts) checks.

Reply:
https://github.com/VenusProtocol/isolated-pools/pull/418